torch/utils/_benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -0,0 +1,448 @@
+            else:
+                first, second = self.stmt_exclusive_stats, other.stmt_exclusive_stats
+        return self._diff(first, second)
+
+    # FIXME: Once 3.7 is the minimum version, type annotate output
+    def as_standardized(self):
+        """Strip some prefixes from function strings.
+
+        When comparing two different sets of instruction counts, on stumbling
+        block can be path prefixes. Callgrind includes the full filepath
+        when reporting a function (as it should). However, this can cause
+        issues when diffing profiles. If a key component such as Python
+        or PyTorch was built in separate locations in the two profiles, which
+        can result in something resembling:
+            23234231 /tmp/first_build_dir/thing.c:foo(...)
+             9823794 /tmp/first_build_dir/thing.c:bar(...)
+              ...
+               53453 .../aten/src/Aten/...:function_that_actually_changed(...)
+              ...
+             -9823794 /tmp/second_build_dir/thing.c:bar(...)
+            -23234231 /tmp/second_build_dir/thing.c:foo(...)
+
+        Stripping prefixes can ameliorate this issue by regularizing the
+        strings and causing better cancelation of equivilent call sites
+        when diffing.
+        """
+        def strip_prefix(stats: Tuple[Tuple[int, str], ...]) -> Tuple[Tuple[int, str], ...]:
+            counts: DefaultDict[str, int] = collections.defaultdict(int)
+
+            # "Python" and "Objects" come from CPython.
+            prefix_truncations = ("build/aten/", "Python/", "Objects/")
+            for c, fn in stats:
+                fn = re.sub(r"^.+build/\.\./", "build/../", fn)
+                for new_prefix in prefix_truncations:
+                    fn = re.sub(r"^.+" + new_prefix, new_prefix, fn)
+                fn = re.sub(r"\s\[.+\]$", "", fn)
+                counts[fn] += c
+            return tuple(sorted([(c, fn) for fn, c in counts.items() if c], reverse=True))
+
+        return CallgrindStats(
+            stmt=self.stmt,
+            setup=self.setup,
+            number=self.number,
+            num_threads=self.num_threads,
+            built_with_debug_symbols=self.built_with_debug_symbols,
+            baseline_inclusive_stats=strip_prefix(self.baseline_inclusive_stats),
+            baseline_exclusive_stats=strip_prefix(self.baseline_exclusive_stats),
+            stmt_inclusive_stats=strip_prefix(self.stmt_inclusive_stats),
+            stmt_exclusive_stats=strip_prefix(self.stmt_exclusive_stats),
+        )
+
+    @staticmethod
+    def _counts(stats, include_lookdict_unicode: bool) -> int:
+        return sum(
+            c for c, fn in stats
+            if include_lookdict_unicode
+            or "dictobject.c:lookdict_unicode" not in fn
+        )
+
+    @staticmethod
+    def _diff(first: Tuple[Tuple[int, str], ...], second: Tuple[Tuple[int, str], ...]) -> Tuple[Tuple[int, str], ...]:
```

Comment:
Getting super strong vibes of "You should just have a class that semantically represents `Tuple[Tuple[int, str], ...]` here

Reply:
This feels especially relevant because you keep converting these tuples into defaultdicts and then converting them back into tuples when you're done.